examples/pinned.cpp

```diff
@@ -0,0 +1,76 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC and CHAI
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//////////////////////////////////////////////////////////////////////////////
+#include "../src/util/forall.hpp"
+#include "chai/ManagedArray.hpp"
+
+#include <iostream>
+
```

Comment:
can we add an example / test that uses slices with pinned buffers, just to ensure that all works as expected?